### PR TITLE
Story1task1

### PR DESCRIPTION
**Pull request by Andrei-Trifu**
This ticket has the combined story1task1 and story1task5
- created navigation component 
- some minimal styling
- installed react-router-dom
- stripped all the React extra styling

### CODE REVIEW

File: src/assets/components/Nav/Nav.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+function Nav() {
+  return (
+    <nav className="navigation">
+      <div>
+        <h1>Book Swap</h1>
+      </div>
+      <div className="links">
+        <Link to="availableBooks">Available books</Link>
+        <Link to="claimedBooks">Claimed books</Link>
+        <Link to="addBook">Add book</Link>
```
Comment (GDTimi) [inline src/assets/components/Nav/Nav.jsx]: I think we'll want these 'Link' items to be 'NavLinks', as it adds the .active class to the element for CSS targeting, so we will be able to highlight which section we are on, e.g.:

```
// The Navlink component is the same as a link, but it automatically 
// applies a class of "active" to whichever one is active 
<nav> 
<NavLink to="/">Home</NavLink> 
<NavLink to="about">About</NavLink> 
<NavLink to="blog">Blog</NavLink> 
</nav> 
```

File: src/index.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+* {
+  box-sizing: border-box;
+}
```
Comment (GDTimi) [inline src/index.css]: I'm a bit concerned by the "*" here - won't this set the value for every single element on the page? I think we could take it out if not required, to not have any unintended knock-on effects.

**Review by GDTimi — CHANGES_REQUESTED** [review]
Hey Andrei - think the stripping all looks good, I had a couple of things to look at re CSS targeting and updating Routing Links to NavLinks. Apart from that, I think we can send this through and see if team members can get the changes from the story1 branch after.

File: src/index.css (hunk 2 above)
Comment (Andrei-Trifu) [reply src/index.css]: https://www.w3schools.com/css/css3_box-sizing.asp
Screenshot 2023-10-30 at 13 34 08

File: src/assets/components/Nav/Nav.jsx (hunk 1 above)
Comment (GDTimi) [reply src/assets/components/Nav/Nav.jsx]: Looks good - we may have to update once we have looked at the subsequent stories, to see how the data filtering works for available / claimed books, but think we will do that when we get to those stories.

File: src/index.css (hunk 2 above)
Comment (GDTimi) [reply src/index.css]: Ok, that's fine, we'll keep it in - thanks for the supporting info!

**Review by GDTimi — APPROVED** [review]

**Pull request merged**